src/Schema/Comparator.php

```diff
@@ -178,10 +184,7 @@ private function isAutoIncrementSequenceInSchema($schema, $sequence)
         return false;
     }
 
-    /**
-     * @return bool
-     */
-    public function diffSequence(Sequence $sequence1, Sequence $sequence2)
+    public function compareSequences(Sequence $sequence1, Sequence $sequence2): bool
```

Comment:
After looking at the full picture, I think that the new API is less intuitive than the existing one when it comes to the methods that return a boolean (I didn't realize we have some of those). A method `diff($x, $y): bool` is easy to interpret, it returns whether there is a difference between the values or not, but `compare($x, $y)` returning `false` when the values are equal is odd.

Additionally, the new methods would be nice to have to accommodate `$platform` as a required parameter in https://github.com/doctrine/dbal/pull/4659 but the current version accepts it in the constructor which is good as well. 

Let's keep the PR to its original scope and only deprecate the static method. We'll see if more improvements are necessary once the platform-aware patch lands. Sorry if it takes you more than originally planned.

Reply:
> A method `diff($x, $y): bool` is easy to interpret, it returns whether there is a difference between the values or not, but `compare($x, $y)` returning `false` when the values are equal is odd.

I did not think of it while making the changes, but now you say it, the naming is quite odd indeed.

> Let's keep the PR to its original scope and only deprecate the static method.

OK, I will revert the changes except for `compareSchemas()` and `compare()`.

> Sorry if it takes you more than originally planned.

No worries! Small increments are the way to go.